lib/n3h-mod-dht-fullsync/dht-fullsync.js

```diff
@@ -592,6 +600,23 @@ class DhtBackendFullsync extends AsyncClass {
   }
 
   /**
+   * Create map holding the requested addresses
+   * (entryAddress -> contentB64)
+   */
+  async _gatherDataMap (entryAddressList) {
+    const out = new Map()
+    for (let entryAddress of entryAddressList) {
+      // For each entryAddress, send a fetchData event to owner
+      // should receive {entry, metas}
+      let entryWithMeta = await this._fetchDataLocal(entryAddress)
+      log.t('_gatherDataMap entryWithMeta', Buffer.byteLength(entryWithMeta))
+      out.set(entryAddress, entryWithMeta)
```

Comment:
this is going to process serially... we can do it in parallel:

```javascript
const out = new Map()
const wait = []
for (let entryAddress of entryAddressList) {
  wait.push(async () => {
    let entryWithMeta = await this._fetchDataLocal(entryAddress)
    log.t('_gatherDataMap entryWithMeta', Buffer.byteLength(entryWithMeta))
    out.set(entryAddress, entryWithMeta)
  }())
}
await Promise.all(wait)
return out
```

Reply:
Nice, yeah, thanks.
Your code snippet is missing a `()` though (before `async`)